js/src/blocks/components/label.js

```diff
@@ -0,0 +1,35 @@
+// Copied from WooCommerce core and simplified to leave only the needed parts.
```

Comment:
💅  I assume it will be done in future PRs. I expected to have some tests for the JS part as well

Reply:
Only using jest to test this component or custom blocks would be fragile as they involve a lot of Woo core mechanisms and codes. I planned to test them via E2E tests after all required development settled down.